src/apps/external_executor/grpc.h

```diff
@@ -79,65 +137,60 @@ namespace ccf::grpc
 
   template <typename Out>
   void set_grpc_response(
-    const Out& resp, const std::shared_ptr<ccf::RpcContext>& ctx)
+    const GrpcAdapterResponse<Out>& r,
+    const std::shared_ptr<ccf::RpcContext>& ctx)
   {
-    size_t r_size = resp.ByteSizeLong();
-    std::vector<uint8_t> r(r_size);
-
-    auto r_data = r.data();
-
-    auto request_content_type =
-      ctx->get_request_header(http::headers::CONTENT_TYPE);
-    if (request_content_type == http::headervalues::contenttype::GRPC)
+    auto r_ = std::get_if<SuccessResponse<Out>>(&r);
```

Comment:
`r_` is not a great name, and reusing it for different types will bite us. Can we have this as `auto success_response = ...`, and line 170 as `auto error_reponse = ...`?

Reply:
Indeed. I've renamed these.